destreamer.ts

```diff
@@ -91,37 +92,32 @@ async function DoInteractiveLogin(username?: string): Promise<Session> {
     const page = (await browser.pages())[0];
     console.log('Navigating to microsoftonline.com login page...');
 
-    // This breaks on slow connections, needs more reliable logic
-    await page.goto('https://web.microsoftstream.com', { waitUntil: "networkidle2" });
+    await page.goto(loginUrl, { waitUntil: "load" });
     await page.waitForSelector('input[type="email"]');
-    
-    if (username) {
+    if (username){
         await page.keyboard.type(username);
         await page.click('input[type="submit"]');
     }
 
-    await browser.waitForTarget(target => target.url().includes('microsoftstream.com/'), { timeout: 90000 });
-    console.log('We are logged in.');
-    // We may or may not need to sleep here.
-    // Who am i to deny a perfectly good nap?
-    await sleep(1500);
+    // Going trough "remember me" pagin (choosing NO)

```

Comment:
i think this could break some logins
for ex. my login is ms -> other -> ms (now you see "remember me")

i don't want puppeteer to push buttons when i m in "other" login page

Reply:
Ok we can check when we reach the "remember-me" page since it has it's unique URL and then trigger the button press

Reply:
The remember me page is not guaranteed to show up, some tenants have it disabled by policy. Just trust me on that.

Pupeteer should not try to do anything else than maybe fill in the username if specified as an argument. Anything else is outside our control so we should just wait for the browser to land on microsoftstream.com..which i guess is the current behavior, right?

Reply:
Never mind, I was wrong about the url but we can use the input id
An alternative is that we can use the xpath to the element so it can be more adaptive to changes to the DOM

What do you think @kylon ?

Reply:
ok didn't see your comment until i reloaded, going to implement it your way, you are right @snobu